File: solutions/erlang/raindrops/1/src/raindrops.erl
-module(raindrops).

-export([convert/1]).


pling(N) when N rem 3 == 0 -> "Pling";
pling(_)                   -> "".

plang(N) when N rem 5 == 0 -> "Plang";
plang(_)                   -> "".

plong(N) when N rem 7 == 0 -> "Plong";
plong(_)                   -> "".

convert(Number) ->
  Sounds = pling(Number) ++ plang(Number) ++ plong(Number),
  case Sounds of
    "" -> integer_to_list(Number);
    _  -> Sounds
  end.

File: solutions/haskell/raindrops/1/src/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n = if null str then show n else str
  where
    factor (x,y) = if n `mod` x == 0 then y else []
    sounds = [(3,"Pling"), (5,"Plang"), (7,"Plong")]
    str = concatMap factor sounds
